src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -460,11 +465,20 @@ public void processRequest(Request request) {
             if (request.type == OpCode.closeSession) {
                 cnxn.sendCloseSession();
             }
+            checkLatency(request);
         } catch (IOException e) {
             LOG.error("FIXMSG",e);
         }
     }
 
+    private void checkLatency(Request request) {
+        long requestLatency = Time.currentElapsedTime() - request.createTime;
+        if((requestWarnThresholdMs > -1 && requestLatency > requestWarnThresholdMs) || requestWarnThresholdMs == 0) {
```

Comment:
@karanmehta93 really *really* minor nit: 
```
boolean enabledAndAboveThreshold = requestWarnThresholdMs == 0 || (requestWarnThresholdMs > -1 && requestLatency > requestWarnThresholdMs);

if (enabledAndAboveThreshold) {
```

I moved `requestWarnThresholdMs == 0` to the beginning of the expression so that the whole expression can be short circuited if it is equals zero. ;-)

But this is a so trivial recommendation that it's up to you use it or not.